README.md

```diff
@@ -1,42 +1,108 @@
 # How it works
 
-The th2 Codec component is responsible for encoding and decoding the messages. It is operating two instances of encoder/decoder pairs, one for operational purposes and one for general conversion.
+The th2 Codec component is responsible for encoding and decoding the messages.
+It operates two instances of encoder/decoder pairs, one is used for operational purposes and the other for general conversion.
 
-Encoding and decoding is performed according to the scheme "one input queue and one or more output queues". Output queues can have filters, depending on which the output of the encoder and decoder can be partially or completely filtered out. The metadata of the message and its fields are used as filter parameters.
+Encoding and decoding perform according to the scheme "one or more input pins and one or more output pins".
+Both types of pins may have filters. The input / output of the encoder and decoder can be partially or entirely filtered out depending on which filters the pin has.
+The metadata of the message and its fields can be used as filter parameters.
 
-One instance of the codec implements the logic for encoding and decoding one protocol of one version. Version-specific protocol messages are described in a separate xml file called a "dictionary".
+One instance of the codec implements the logic for encoding and decoding one protocol of one version.
+The version-specific protocol messages are described in a separate XML file called "dictionary".
 Codec operates with arrays of messages (parsed batch to raw batch in case of encoding and raw batch to parsed batch upon decoding).
 
+## Appointment
+
+This project is only an adapter between the Sailfish and the th2 packed into Docker Image.
+This [th2-codec-generic](https://github.com/th2-net/th2-codec-generic) project uses this image as a base.
+
 # Running
 
-To start a codec it is required to place external codec api implementation jar with dependencies to 'codec_implementation' folder.
+The codec requires an implementation of the external codec API.
+The JAR file with that implementation and all its dependencies need to be placed to the folder `home/codec_implementation`, in order to start the codec.
+
+The codec loads all JAR files from that directory and looks for all implementations of
```

Comment:
The codec loads all JAR files from that directory and looks for all **the** implementations of